### PR TITLE
Add --condor-requirements option for work_queue_factory. 

### PR DESCRIPTION
**Pull request by charleszheng44**
Resolve Issue #1103, Add --condor-requirements option for work_queue_factory 
with "-T condor" option, user can also specify the
requirements for desired nodes in condor pool.

	eg. work_queue_submit  -M project_name
		-T condor
                --condor-requirements '(MachineGroup == "disc")'

### CODE REVIEW

File: batch_job/src/batch_job_condor.c [hunk 1]
```diff
@@ -128,10 +128,19 @@ static batch_job_id_t batch_job_condor_submit (struct batch_queue *q, const char
 	fprintf(file, "keep_claim_idle = 30\n");
 	fprintf(file, "log = %s\n", q->logfile);
 
+	const char *c_req = batch_queue_get_option(q, "condor-requirements");
 	char *bexp = blacklisted_expression(q);
```
Comment (dthain) [inline batch_job/src/batch_job_condor.c]: Take a closer look at this, do you see where bexp is allocated and freed?

File: batch_job/src/batch_job_condor.c [hunk 2]
```diff
@@ -128,10 +128,19 @@ static batch_job_id_t batch_job_condor_submit (struct batch_queue *q, const char
...
-	if(bexp) {
-		fprintf(file, "requirements = %s\n", bexp);
-		free(bexp);
+    if(c_req) {
```
Comment (btovar) [inline batch_job/src/batch_job_condor.c]: I don't think this quite works. If c_req is NULL, bexp is never used.

Comment (charleszheng44) [inline batch_job/src/batch_job_condor.c]: Am I supposed to use the `indent` program?

Comment (btovar) [inline batch_job/src/batch_job_condor.c]: The autobuild email you got had some advice on how to deal with this:

Checking suitability for merging:
        Whitespace check... failed.
                Fix using: git rebase --whitespace=fix $(git merge-base master HEAD)
                git diff --check master...c4de028f5e201fed27cc3d5b2d627389ec482c71
                        batch_job/src/batch_job_condor.c:133: indent with spaces.
                        +    if(c_req) {
                        batch_job/src/batch_job_condor.c:134: indent with spaces.
                        +        if(bexp) {
                        batch_job/src/work_queue_factory.c:726: indent with spaces.
                        +            case LONG_OPT_CONDOR_REQUIREMENTS:
                        batch_job/src/work_queue_factory.c:728: indent with spaces.
                        +                // will implement
                        batch_job/src/work_queue_factory.c:729: space before tab in indent.
                        +                               break;
                        batch_job/src/work_queue_factory.c:845: indent with spaces.
                        +    if (condor_requirements != NULL) {
                        batch_job/src/work_queue_factory.c:847: indent with spaces.
                        +    }

Comment (dthain) [inline batch_job/src/batch_job_condor.c]: Try the cctools.indent script that is in the cctools repository.

**Comment by charleszheng44** [comment]
1. Standardize the whitespace and fix the memory leak problem.
2. Collapse two commits into one.

**Comment by btovar** [comment]
@charleszheng44: please also make condor-requirements an option in the factory configuration file.

**Comment by charleszheng44** [comment]
* Add condor-requirements to factory configuration file
* Resolve issue #1138

**Pull request closed without merging**